airflow/hooks/presto_hook.py

```diff
@@ -117,11 +120,63 @@ def get_pandas_df(self, hql, parameters=None):
             df = pandas.DataFrame()
         return df
 
-    def run(self, hql, parameters=None):
+    def run(self, sql, parameters=None, poll_interval=None):
         """
-        Execute the statement against Presto. Can be used to create views.
+        Execute statement(s) against Presto. By default, statements are
+        executed asynchronously. To execute each synchronously, pass a non-None
+        poll_interval.
+
+        :param sql: the statement(s) to be executed
+        :type sql: str or iterable
+        :param parameters: the parameters to render the statement(s) with
+        :type parameters: mapping or iterable
+        :param poll_interval: how often, in seconds, to check the execution
+            status of each statement; set to None
+        :type poll_interval: int or float
         """
-        return super(PrestoHook, self).run(self._strip_sql(hql), parameters)
+        if isinstance(sql, past.builtins.basestring):
+            sql = [sql]
+
+        cursor = self.get_conn().cursor()
+
+        for stmt in sql:
+            stmt = self._strip_sql(stmt)
+            self.log.info("{} with parameters {}".format(stmt, parameters))
+            cursor.execute(stmt, parameters)
+
+            if poll_interval is not None:
+                while not self.execution_finished(cursor):
```

Comment:
It's definitely A Bad Idea™️  to poll indefinitely, but I'm assuming all the timeout logic is expected to live in the operators using this hook. @Fokko please LMK if that's assumption's inaccurate, and if there ought to be some minimal safeguards here, e.g. an upper bound on the number of pings to send or time to wait.

Reply:
In other operators there are similar constructions that define an upper bound to thrown an exception when the maximum time is exceeded: https://github.com/apache/airflow/blob/master/airflow/hooks/druid_hook.py#L90